lib/types/object/index.js

```diff
@@ -857,8 +858,9 @@ internals.or = function (value, peers, parent, state, options) {
 
     for (let i = 0; i < peers.length; ++i) {
         const peer = peers[i];
-        if (Object.prototype.hasOwnProperty.call(parent, peer) &&
-            parent[peer] !== undefined) {
+        const splitPeers = peer.split('.');
+        const keysExist = Utils.objHasKeys(parent, splitPeers);
```

Comment:
You can certainly use `Hoek.reach` to achieve the same thing instead of creating your own.

Reply:
Oh! I'll check it out and update the PR. Thanks @Marsup

Reply:
Done @Marsup